confidant/routes/credentials.py

```diff
@@ -1126,3 +1126,69 @@ def generate_value():
     if len(value) > VALUE_LENGTH:
         value = value[:VALUE_LENGTH]
     return jsonify({'value': value})
+
+
+@blueprint.route('/v1/credentials/<id>/archive', methods=['POST'])
+@authnz.require_auth
+@authnz.require_csrf_token
+@maintenance.check_maintenance_mode
+def archive_credential(id):
+    '''
+    Archive the provided credential.
+    **Example request**:
+
+    .. sourcecode:: http
+
+       POST /v1/credentials/abcd12345bf4f1cafe8e722d3860404/archive
+
+    :param id: The credential ID to update.
+    :<json bool force: Run the archive operation, if false will only validate
+        the provided input
+    **Example response**:
+
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+        {
+          "result": "success"
+        }
+
+    :resheader Content-Type: application/json
+    :statuscode 200: Success
+    :statuscode 400: Invalid input or state
+    :statuscode 403: Client does not have access to archive the provided
+                     credential ID.
+    '''
+    if not acl_module_check(resource_type='credential',
+                            action='archive',
+                            resource_id=id):
+        msg = "{} does not have access to archive credential {}".format(
+            authnz.get_logged_in_user(),
+            id
+        )
+        error_msg = {'error': msg, 'reference': id}
+        return jsonify(error_msg), 403
+
+    try:
+        _cred = Credential.get(id)
+    except DoesNotExist:
+        return jsonify({'error': 'Credential not found.'}), 404
+    if _cred.data_type != 'credential':
+        msg = 'id provided is not a credential.'
+        return jsonify({'error': msg}), 400
+
+    data = request.get_json()
+    force = data.get('force') if data else False
```

Comment:
```suggestion
    force = data.get('force', False) if data else False
```

Reply:
dbl check me on this (i haven't ran any code), but we basically wanna fail safe and force explicitly setting the boolean to True if we want to archive.